src/Crypto.gren

```diff
@@ -0,0 +1,2227 @@
+import Debug
+import Bytes exposing ( Bytes )
+import Bytes.Encode
+import Maybe exposing ( Maybe(..) )
+import Result exposing ( Result(..) )
+import Set exposing ( Set )
+import Bitwise
+import Math
+import Json.Encode
+import Gren.Kernel.Json
+
+
+
+-- RANDOM UUID
+
+
+{-| Generate a random UUID using the UUID v4 algorithm.
+-}
+randomUuidV4 : SecureContext -> Task x String
+randomUuidV4 _context =
+    Gren.Kernel.Crypto.randomUUID
+
+
+
+-- RANDOM VALUES
+
+
+{-| Get some `Bytes` of random, signed, 8-bit values equal to the length of the passed `Int`
+with a maximum of 65536 values.
+-}
+getRandomInt8Values : Int -> Task x Bytes
+getRandomInt8Values int =
+    Gren.Kernel.Crypto.getRandomValues (clamp 0 65536 int) "int8"
+
+
+{-| Get some `Bytes` of random, unsigned, 8-bit values equal to the length of the passed `Int`
+with a maximum of 65536 values.
+-}
+getRandomUInt8Values : Int -> Task x Bytes
+getRandomUInt8Values int =
+    Gren.Kernel.Crypto.getRandomValues (clamp 0 65536 int) "uint8"
+
+ 
+{-| Get some `Bytes` of random, signed, 16-bit values equal to the length of the passed `Int`.
+with a maximum of 32768 values.
+-}
+getRandomInt16Values : Int -> Task x Bytes
+getRandomInt16Values int =
+    Gren.Kernel.Crypto.getRandomValues (clamp 0 32768 int) "int16"
+
+
+{-| Get some `Bytes` of random, unsigned, 16-bit values equal to the length of the passed `Int`.
+with a maximum of 32768 values.
+-}
+getRandomUInt16Values : Int -> Task x Bytes
+getRandomUInt16Values int =
+    Gren.Kernel.Crypto.getRandomValues (clamp 0 32768 int) "uint16"
+
+
+{-|Get some `Bytes` of random, signed, 32-bit values equal to the length of the passed `Int`
+with a maximum of 16384 values.
+-}
+getRandomInt32Values : Int -> Task x Bytes
+getRandomInt32Values int =
+    Gren.Kernel.Crypto.getRandomValues (clamp 0 16384 int) "int32"
+
+
+{-| Get some `Bytes` of random, unsigned, 32-bit values equal to the length of the passed `Int`
+with a maximum of 16384 values.
+-}
+getRandomUInt32Values : Int -> Task x Bytes
+getRandomUInt32Values int =
+    Gren.Kernel.Crypto.getRandomValues (clamp 0 16384 int) "uint32"
+
+
+
+-- ENVIRONMENT
+
+
+{-| Represents the platform being considered secure. This type can be generated using the 
+`getSecureContext` function and is required for some functions to be run in this package.
+-}
+type SecureContext
+    = SecureContext
+
+
+{-| A `Task` that succeeds with `SecureContext` if the code is being run in a 
+secure context. If this `Task` fails, most of the functions within this module
+will not be able to run.
+
+While running this function on the `node` platform, it should succeed unless you
+are running a version of `node` that does not support the Web Crypto APIs.
+
+While running this function on the `browser` platform, it will succeed if the
+application being run is [considered as being in a secure context](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts).
+-}
+getSecureContext : Task x SecureContext
```

Comment:
This task can fail, so the error value should be a proper type. I think `{}` will suffice in this case.
Don't forget to update the kernel code.